diff --git a/libvmaf/src/cuda/picture_cuda.c b/libvmaf/src/cuda/picture_cuda.c
index 835cc7cf..b84ec91c 100644
--- a/libvmaf/src/cuda/picture_cuda.c
+++ b/libvmaf/src/cuda/picture_cuda.c
@@ -229,7 +229,7 @@ int vmaf_cuda_picture_free(VmafPicture *pic, void *cookie)
 
     for (int i = 0; i < 3; i++) {
 	    if (pic->data[i]) {
-	        CHECK_CUDA(cu_f ,cuMemFreeAsync((CUdeviceptr)pic->data[i], priv->cuda.str));
+	        CHECK_CUDA(cu_f ,cuMemFree((CUdeviceptr)pic->data[i]));
 	    }
     }
 
